instructure/instructure-ui
# required asterisk and new form error messages

INSTUI-4321

test plan:

- please go through the code thoroughly to check the implementation details
- try to manually reproduce the new error state in multiple components based on the design docs
- check the TextInput readme for a showcase of all the different components and states
- check the [new form error guide](https://instructure.github.io/instructure-ui/pr-preview/pr-1729/#form-errors)
